Forta-Bots-Challenge-2/src/utils.ts

```diff
@@ -0,0 +1,75 @@
+import { ethers } from "ethers";
+import { Provider } from "@ethersproject/providers";
+import { LRUCache } from "lru-cache";
+import { POOL_INIT_CODE_HASH } from "./constants";
+
+export const poolAddressCache: LRUCache<string, boolean> = new LRUCache({
+  max: 1000,
+});
+
+export const getPoolValues = async (
+  poolAddress: string,
+  provider: ethers.providers.Provider,
+  uniswapPoolABI: string[],
+  blockNumber: number,
+) => {
+  const uniswapPoolContract = new ethers.Contract(
+    poolAddress,
+    uniswapPoolABI,
+    provider,
+  );
+  // Use Promise.all to parallelize the asynchronous calls
+  const [token0, token1, fee] = await Promise.all([
+    uniswapPoolContract.token0({ blockTag: blockNumber }),
+    uniswapPoolContract.token1({ blockTag: blockNumber }),
+    uniswapPoolContract.fee({ blockTag: blockNumber }),
+  ]);
+
+  return { token0, token1, fee };
+};
+
+export const computeAddress = (
+  factoryAddress: string,
+  poolValues: any[],
+  poolHash: string,
+) => {
+  const encodedAbiData = ethers.utils.defaultAbiCoder.encode(
+    ["address", "address", "uint24"],
+    poolValues,
+  );
+  const derivationIdentifier = ethers.utils.solidityKeccak256(
+    ["bytes"],
+    [encodedAbiData],
+  );
+  const poolAddress = ethers.utils.getCreate2Address(
+    factoryAddress,
+    derivationIdentifier,
+    poolHash,
+  );
+  return poolAddress;
+};
+export const isUniswapAddress = async (
```

Comment:
this requires 7 arguments but in agent.ts you provide 5, which blocks the bot to run. please remove the unused arguments